web/src/js/ducks/settings.js

```diff
@@ -1,64 +1,72 @@
-import { addLogEntry } from './eventLog'
+import {fetchApi} from "../utils";
 
-export const WS_MSG_TYPE = 'settings'
-export const WS_MSG_CMD_UPDATE = 'update'
+export const REQUEST_SETTINGS = "REQUEST_SETTINGS"
+export const RECEIVE_SETTINGS = "RECEIVE_SETTINGS"
+export const UPDATE_SETTINGS = "UPDATE_SETTINGS"
 
-export const BEGIN_FETCH = 'SETTINGS_BEGIN_FETCH'
-export const FETCHED = 'SETTINGS_FETCHED'
-export const RECV_WS_MSG = 'SETTINGS_RECV_WS_MSG'
-
-const defaultState = { settings: {}, pendings: null }
+const defaultState = {
+    settings: {},
+    isFetching: false,
+    actionsDuringFetch: [],
+}
 
-export default function reduce(state = defaultState, action) {
+export default function reducer(state = defaultState, action) {
     switch (action.type) {
 
-        case BEGIN_FETCH:
-            return { ...state, pendings: [] }
+        case REQUEST_SETTINGS:
+            return {
+                ...state,
+                isFetching: true
+            }
 
-        case FETCHED:
-            const pendings = state.pendings || []
-            return { ...state, pendings: null, settings: pendings.reduce(reduceData, action.data) }
+        case RECEIVE_SETTINGS:
+            let s = {
+                settings: action.settings,
+                isFetching: false,
+                actionsDuringFetch: [],
+            }
+            for (action of state.actionsDuringFetch) {
+                s = reducer(s, action)
+            }
+            return s
```

Comment:
I considered that as well, but I believe the for loop is more readable (see: http://stackoverflow.com/a/181706)